sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -163,7 +169,12 @@ function getRequestBody(body?: unknown, contentType: string = ""): RequestBody {
   }
 
   if (!contentType && typeof body === "string") {
-    return { body };
+    try {
+      JSON.parse(body);
+      return { body };
+    } catch (error) {
+      return { body: JSON.stringify(body) };
+    }
```

Comment:
try to parse the body string if no content type specified, if it's a valid json string, then we should just return the original string, if it's not a valid json string, we should return the stringified string.

Reply:
This feels strange to me, like if the endpoint takes a text/plain we'd encode it like it took JSON unless the text/plain got passed through in the content-type header? I'm not sure why we should be assuming the request body is a JSON

Reply:
Actually, we have a function to set the default content type as application/json here https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client-rest/src/sendRequest.ts#L94-L107, however we don't really use that when we build the request body so more of an inconsistency inside the core-client-rest itself.